src/app/static/src/app/store/modelOutput/actions.ts

```diff
@@ -13,31 +14,45 @@ export const actions: ActionTree<ModelOutputState, DataExplorationState> & Model
     async updateSelectedTab(context, tab) {
         const {commit, rootState, dispatch} = context;
 
-        const currentIndicators: string[] = [];
+        const currentIndicatorPayloads: ResultDataPayload[] = [];
         switch (tab) {
             case ModelOutputTabs.Bar:
-                currentIndicators.push(rootState.plottingSelections.barchart.indicatorId);
+                currentIndicatorPayloads.push({
+                    tab,
+                    payload: { indicatorId: rootState.plottingSelections.barchart.indicatorId }
+                });
                 break;
             case ModelOutputTabs.Bubble:
-                currentIndicators.push(rootState.plottingSelections.bubble.colorIndicatorId);
-                currentIndicators.push(rootState.plottingSelections.bubble.sizeIndicatorId);
+                currentIndicatorPayloads.push({
+                    tab,
+                    payload: { colorIndicatorId: rootState.plottingSelections.bubble.colorIndicatorId }
+                });
+                currentIndicatorPayloads.push({
+                    tab,
+                    payload: { sizeIndicatorId: rootState.plottingSelections.bubble.sizeIndicatorId }
```

Comment:
I'm a bit confused about why you're making multiple payloads here, when the `getResultData` method has the ability to handle both color and size indicators in a single payload.